src/github/types/config.ts

```diff
@@ -0,0 +1,25 @@
+import { Type as T } from "@sinclair/typebox";
+import { StaticDecode } from "@sinclair/typebox";
+import { GitHubEvent } from "./github-events";
+
+enum Commands {
+  Start = "start",
+  Stop = "stop",
+}
+
+const handlerSchema = T.Array(
+  T.Object({
+    workflow: T.String(),
+    settings: T.Unknown(),
+  }),
+  { default: [] }
+);
+
+export const configSchema = T.Object({
+  handlers: T.Object({
+    commands: T.Record(T.Enum(Commands), handlerSchema, { default: {} }),
+    events: T.Record(T.Enum(GitHubEvent), handlerSchema, { default: {} }),
```

Comment:
I'd like to avoid using our own enum for events but the problem is that octokit provides only a readonly array of all events or a typescript union type of all events but I haven't figured out how to use them with typebox

Reply:
An out of the box idea is to generate the enum file dynamically based on the dependency contents at build time. 

Given the time constraints for getting this stable, we can consider breaking that task off into a separate issue.

Reply:
I don't get how you can use `eventsEnum` with Typebox? Can you provide an example?

For reference I've already tried making Union from Literals like this:
```
const eventNames = emitterEventNames.map((value) => T.Literal(value));
...
events: T.Record(T.Union(eventNames), handlerSchema, { default: {} })
```
and while this does work, the typescript type of `events` becomes just `{}`

Reply:
Sorry @whilefoo probably useless now seeing as this was all the way back in February, please tag me again if this ever happens. As far as I understand to get enums to work you just build a object of `key:key` as that's what a TS enum boils down to in JS essentially and then use `as const` on the object and feed it to `T.Enum()`. I don't use TS enums myself so I'm not sure how to get those to work